src/Controllers/API/AddCourseCategoryController.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Portal\Controllers\API;
+
+use Portal\Abstracts\Controller;
+use Portal\Sanitisers\CourseCategorySanitiser;
+use Psr\Http\Message\ServerRequestInterface as Request;
+use Psr\Http\Message\ResponseInterface as Response;
+use Portal\Models\CourseModel;
+
+class AddCourseCategoryController extends Controller
+{
+    private CourseModel $courseModel;
+
+    public function __construct(CourseModel $courseModel)
+    {
+        $this->courseModel = $courseModel;
+    }
+
+    /**
+     * Get user input and send to CourseModel to add to db.
+     */
+    public function __invoke(Request $request, Response $response, array $args): Response
+    {
+        $newCategory = $request->getParsedBody();
+
+        $responseData = [
+            'success' => false,
+            'message' => 'Unexpected Error.',
+            'data' => []
+        ];
+        $statusCode = 400;
+
+        try {
+                $newCategory = CourseCategorySanitiser::sanitise($newCategory);
+                $insertedId = $this->courseModel->addCategory($newCategory);
+        } catch (\Exception $exception) {
+            $responseData['message'] = $exception->getMessage();
```

Comment:
if the query fails because the DB crashed, this will send a 400 status code but with a DB broken error, which makes no sense. you need a catch for PDO exception, to set the status to 500, and a catch for the sanitise exceptions to keep the 400 status code